File: src/lessons/bat/string1/StringTimesEntity.java
package lessons.bat.string1;

import static plm.core.ValueSerializer.*;

import plm.universe.bat.BatEntity;

public class StringTimesEntity extends BatEntity {

  public void run()
  {
    int count = getTestCount();
    for (int i = 0; i < count; i++) {
      Object[] param = (Object[])deserialize(getTest(i));
      setTestResult(i, serialize(stringTimes((String)param[0], (Integer)param[1])));
    }
  }

  /* BEGIN TEMPLATE */
  String stringTimes(String str, int n)
  {
    /* BEGIN SOLUTION */
    String result = "";
    for (int i = 0; i < n; i++) {
      result = result + str; // could use += here
    }
    return result;
    /* END SOLUTION */
  }
  /* END TEMPLATE */
}

File: src/lessons/bat/string1/StringTimesEntity.scala
package lessons.bat.string1

import plm.core.ValueSerializer
import plm.universe.bat.BatEntity

class StringTimesEntity extends BatEntity {

    override def run() {
      import ValueSerializer._

      val count = getTestCount()
      for (i <- 0 to count -1) {
        val param = deserialize(getTest(i)).asInstanceOf[Array[Object]]
        setTestResult(i, serialize(stringTimes(param(0).asInstanceOf[String], param(1).asInstanceOf[Int])) )
	  }
	}

	/* BEGIN TEMPLATE */
	def stringTimes(str:String, n:Int):String = {
		/* BEGIN SOLUTION */
	  var res = ""
	  for (i <- 1 to n)
	    res ++= str
	  return res
		/* END SOLUTION */
	}
	/* END TEMPLATE */
}
